### PR TITLE
[fix](auth)Http api check auth

### PR DESCRIPTION
**Pull request by zddr**
- add auth for `api/meta/namespaces/internal/databases`
- add auth for `api/show_table_data`

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run performance

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/MetaInfoAction.java [hunk 1]
```diff
@@ -105,18 +105,18 @@ public Object getAllDatabases(
...
-                    .checkDbPriv(ConnectContext.get(), InternalCatalog.INTERNAL_CATALOG_NAME, fullName,
+                    .checkDbPriv(ConnectContext.get(), ns, fullName,
                             PrivPredicate.SHOW)) {
                 continue;
             }
             dbNameSet.add(db);
         }
 
-        Collections.sort(dbNames);
+        Collections.sort(dbNameSet);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/MetaInfoAction.java:115]: rename to `filteredDbNames`

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/ShowAction.java [hunk 2]
```diff
@@ -212,18 +214,25 @@ public Object show_data(HttpServletRequest request, HttpServletResponse response
 
     @RequestMapping(path = "/api/show_table_data", method = RequestMethod.GET)
     public Object show_table_data(HttpServletRequest request, HttpServletResponse response) {
-        if (Config.enable_all_http_auth) {
-            executeCheckPassword(request, response);
-            checkGlobalAuth(ConnectContext.get().getCurrentUserIdentity(), PrivPredicate.ADMIN);
-        }
+        executeCheckPassword(request, response);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/ShowAction.java:217]: keep `if (Config.enable_all_http_auth)`

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/ShowAction.java [hunk 3]
```diff
@@ -212,18 +214,25 @@ public Object show_data(HttpServletRequest request, HttpServletResponse response
...
 
         String dbName = request.getParameter(DB_KEY);
         String tableName = request.getParameter(TABLE_KEY);
+
+        if (StringUtils.isEmpty(dbName) && StringUtils.isEmpty(tableName)) {
+            return ResponseEntityBuilder.okWithCommonError("dbName and tableName cannot be empty at the same time");
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/ShowAction.java:223]:
```suggestion
            return ResponseEntityBuilder.okWithCommonError("db and table cannot be empty at the same time");
```

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/ShowAction.java [hunk 4]
```diff
@@ -212,18 +214,25 @@ public Object show_data(HttpServletRequest request, HttpServletResponse response
...
+        }
+
         String singleReplica = request.getParameter(SINGLE_REPLICA_KEY);
         boolean singleReplicaBool = Boolean.parseBoolean(singleReplica);
         Map<String, Map<String, Long>> oneEntry = Maps.newHashMap();
+        UserIdentity user = ConnectContext.get().getCurrentUserIdentity();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/ShowAction.java:229]: Add `if (Config.enable_all_http_auth)`
Only check auth when enable_all_http_auth is true

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/ShowAction.java [hunk 5]
```diff
@@ -236,6 +245,11 @@ public Object show_table_data(HttpServletRequest request, HttpServletResponse re
                 if (db == null || !(db instanceof Database) || ((Database) db) instanceof MysqlCompatibleDatabase) {
                     continue;
                 }
+                if (!Env.getCurrentEnv().getAccessManager()
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/ShowAction.java:248]: ditto

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/ShowAction.java [hunk 6]
```diff
@@ -331,6 +345,12 @@ private Map<String, Long> getDataSizeOfTables(DatabaseIf db, String tableName, b
             if (Strings.isNullOrEmpty(tableName)) {
                 List<Table> tables = db.getTables();
                 for (Table table : tables) {
+                    if (!Env.getCurrentEnv().getAccessManager()
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/ShowAction.java:348]: ditto

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by morrySnow — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**